From f828f106e0ca1e1d110bf280953f9a677af785a9 Mon Sep 17 00:00:00 2001
From: Zhen Chen <chenzhen@rock-chips.com>
Date: Tue, 27 Aug 2019 15:37:10 +0800
Subject: [PATCH] MALI: bifrost: only call dma_fence_set_error() for actual
 errors

This would eliminate the repeating WARNING logs from dma_fence_set_error().

Change-Id: I3b319decb8899dfc0410f4975e31bb1c67f7ae9f
Signed-off-by: Zhen Chen <chenzhen@rock-chips.com>
---
 drivers/gpu/arm/bifrost/mali_kbase_fence.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/arm/bifrost/mali_kbase_fence.h b/drivers/gpu/arm/bifrost/mali_kbase_fence.h
index 414f3f4d7436..5c841b186ed5 100644
--- a/drivers/gpu/arm/bifrost/mali_kbase_fence.h
+++ b/drivers/gpu/arm/bifrost/mali_kbase_fence.h
@@ -143,7 +143,8 @@ static inline int kbase_fence_out_signal(struct kbase_jd_atom *katom,
 	  KERNEL_VERSION(4, 9, 68) <= LINUX_VERSION_CODE)
 	fence_set_error(katom->dma_fence.fence, status);
 #elif (KERNEL_VERSION(4, 11, 0) <= LINUX_VERSION_CODE)
-	dma_fence_set_error(katom->dma_fence.fence, status);
+	if (status < 0)
+		dma_fence_set_error(katom->dma_fence.fence, status);
 #else
 	katom->dma_fence.fence->status = status;
 #endif
-- 
2.35.3

